src/theory/strings/extf_solver.cpp

```diff
@@ -90,7 +90,9 @@ bool ExtfSolver::shouldDoReduction(int effort, Node n, int pol)
     Trace("strings-extf-debug") << "...skip due to model active" << std::endl;
     return false;
   }
-  if (d_reduced.find(n)!=d_reduced.end())
+  // check with negation
```

Comment:
```suggestion
  // check with negation if requested (only applied to Boolean terms)
  Assert(n.getType().isBoolean() || pol != -1);
```

To make it clear that `n` does not have to be Boolean, but that we also do not generate invalid terms here.